File: 백준/Silver/2579. 계단 오르기/계단 오르기.py
N = int(input())
point = [int(input()) for _ in range(N)]

dp = [0] * (N + 1)
dp[1] = point[0]

if N >= 2:
    dp[2] = point[0] + point[1]

for i in range(3, N + 1):
    dp[i] = max(dp[i-3] + point[i-2] + point[i-1], dp[i-2] + point[i-1])

print(dp[N]);

File: 백준/Silver/2579. 계단 오르기/계단 오르기.js
const input = require('fs').readFileSync(0).toString().trim().split('\n');
const N = Number(input.shift());
const points = input.map(Number);

const dp = new Array(N + 1).fill(0);
dp[1] = points[0];

if (N >= 2) {
    dp[2] = points[0] + points[1];
}

for (let i = 3; i <= N; i++) {
    dp[i] = Math.max(dp[i-3] + points[i-2] + points[i-1], dp[i-2] + points[i-1]);
}

console.log(dp[N]);